### PR TITLE
skip trace.json file reader for tf 2.11 and above since it's no longe…

### PR DESCRIPTION
**Pull request by ztlevi**
### Description of changes:
TF 2.11 no longer have trace.json.gz file available. They merged all data to `xxx-.xplane.pb` file. This PR disables TF framework profiler support for TF 2.11 and above to unblock the AWS TF release.

#### Style and formatting:

I have run `pre-commit install && pre-commit run --all-files` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/profiler/core/test_utils.py [hunk 1]
```diff
@@ -138,7 +142,8 @@ def test_timeline_merge_by_step(out_dir, trace_location):
 
     assert total_events == num_trace_events
 
-
+@pytest.mark.skipif(version.parse(tf.__version__) > version.parse("2.10"),
+                    reason="tf 2.11 merged trace.json file to xplane.pb file.")
```
Comment (Roshrini) [inline tests/profiler/core/test_utils.py:146]: why do you need to skip all tests related to merging timelines? only one file is not available, rests still should work

**Review by MZSHAN — APPROVED** [review]

File: smdebug/profiler/algorithm_metrics_reader.py [hunk 2]
```diff
@@ -4,6 +4,10 @@
 import json
 import os
 
+# Third Party
+import tensorflow as tf
```
Comment (yl-to) [inline smdebug/profiler/algorithm_metrics_reader.py:8]: what if this test is running in a pytorch container?

Comment (ztlevi) [reply smdebug/profiler/algorithm_metrics_reader.py]: you're right. reverting the change from this file

**Comment by yl-to** [comment]
If profiler team's decision is to disable the failing tests, I would suggest `@pytest.mark.skipif` before the pytest func def

**Comment by ztlevi** [comment]
Pushed an update for your comments @yl-to

**Comment by yl-to** [comment]
@ztlevi So profiler team's decision is to skip these tests for tensorflow 2.11, please confirm after you tested it.
I will approve the PR once get confirmation.

**Comment by ztlevi** [comment]
Yes, I confirm tests passes.

**Review by yl-to — APPROVED** [review]

**Pull request merged**